nemo/collections/llm/quantization/utils.py

```diff
@@ -24,28 +24,74 @@
 from nemo.utils import logging
 
 
-def get_modelopt_decoder_type(model: llm.GPTModel) -> str:
-    """Infers the modelopt decoder type from GPTModel subclass."""
-    mapping = [
-        (llm.Baichuan2Model, "baichuan"),
-        (llm.ChatGLMModel, "chatglm"),
-        (llm.Gemma2Model, "gemma2"),
-        (llm.GemmaModel, "gemma"),
-        (llm.LlamaModel, "llama"),
-        (llm.MistralModel, "llama"),
-        (llm.MixtralModel, "llama"),
-        (llm.NemotronModel, "gpt"),
-        (llm.Qwen2Model, "qwen"),
-        (llm.StarcoderModel, "gpt"),
-        (llm.Starcoder2Model, "gpt"),
-        (llm.Phi3Model, "phi3"),
-    ]
-
-    for config_class, decoder_type in mapping:
-        if isinstance(model, config_class):
-            return decoder_type
-
-    logging.warning("Could not infer the decoder type")
+huggingface_model_type_pattern_match = {
```

Comment:
Could you tell from where this comes from? Can this be just imported?

Reply:
Maybe we could ask ModelOpt to move it to some importable location if this makes sense @kevalmorabia97 @cjluo-nv